libsrc/hyperion/LinearColorSmoothing.cpp

```diff
@@ -84,28 +84,19 @@ void LinearColorSmoothing::handleSettingsUpdate(const settings::type& type, cons
 
 int LinearColorSmoothing::write(const std::vector<ColorRgb> &ledValues)
 {
+	_targetTime = QDateTime::currentMSecsSinceEpoch() + _settlingTime;
+	_targetValues = ledValues;
+
 	// received a new target color
 	if (_previousValues.empty())
 	{
 		// not initialized yet
-		_targetTime = QDateTime::currentMSecsSinceEpoch() + _settlingTime;
-		_targetValues = ledValues;
-
 		_previousTime = QDateTime::currentMSecsSinceEpoch();
 		_previousValues = ledValues;
 
 		//Debug( _log, "Start Smoothing timer: settlingTime: %d ms, interval: %d ms (%u Hz), updateDelay: %u frames", _settlingTime, _updateInterval, unsigned(1000.0/_updateInterval), _outputDelay );
 		QMetaObject::invokeMethod(_timer, "start", Qt::QueuedConnection, Q_ARG(int, _updateInterval));
 	}
-	else
-	{
-		//std::cout << "LinearColorSmoothing::write> "; LedDevice::printLedValues ( ledValues );
-
-		_targetTime = QDateTime::currentMSecsSinceEpoch() + _settlingTime;
-		memcpy(_targetValues.data(), ledValues.data(), ledValues.size() * sizeof(ColorRgb));
```

Comment:
The reason for choosing ````memcpy```` was probably performance concern but most compilers are clever enough to use ````memove```` if the type is POD.

Reply:
Given that the smoothing function is part of the critical processing path, I would follow the conclusion that the motivation may had been to reduce processing overhead as much as possible. Some of the hyperion users still use rpi 1 / Zero Hardware...
But I would agree to you that normally Qt copies are  implemented efficiently.

Reply:
This one should also have the vectorization possibility, which is not the case for memcpy.

Reply:
Sure. Do not misunderstand my comment, I am in support of the proposed change. Just wanted to highlight that not powerful hardware requires still to be considered:).